broker-core/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java

```diff
@@ -306,25 +305,28 @@ public boolean isClosed() {
 
   private static class RecordExporter {
 
+    private final RecordValueCache recordValueCache = new RecordValueCache();
     private final RecordMetadata rawMetadata = new RecordMetadata();
     private final List<ExporterContainer> containers;
-    private final int partitionId;
+    private final TypedEventImpl typedEvent;
 
-    private Record record;
     private boolean shouldExport;
     private int exporterIndex;
 
     RecordExporter(List<ExporterContainer> containers, int partitionId) {
       this.containers = containers;
-      this.partitionId = partitionId;
+      typedEvent = new TypedEventImpl(partitionId);
     }
 
     void wrap(LoggedEvent rawEvent) {
       rawEvent.readMetadata(rawMetadata);
 
-      shouldExport = EVENT_REGISTRY.containsKey(rawMetadata.getValueType());
+      final UnifiedRecordValue recordValue =
+          recordValueCache.readRecordValue(rawEvent, rawMetadata.getValueType());
+
+      shouldExport = recordValue != null;
```

Comment:
Is `null` a valid return value? I think not since it has one record value for each value type.